mesonbuild/dependencies/misc.py

```diff
@@ -574,3 +576,50 @@ def get_methods(self):
             return [DependencyMethods.PKGCONFIG, DependencyMethods.EXTRAFRAMEWORK]
         else:
             return [DependencyMethods.PKGCONFIG]
+
+class PcapDependency(ExternalDependency):
+    def __init__(self, environment, kwargs):
+        super().__init__('pcap', environment, None, kwargs)
+        if DependencyMethods.PKGCONFIG in self.methods:
+            try:
+                kwargs['required'] = False
+                pcdep = PkgConfigDependency('pcap', environment, kwargs)
+                if pcdep.found():
+                    self.type_name = 'pkgconfig'
+                    self.is_found = True
+                    self.compile_args = pcdep.get_compile_args()
+                    self.link_args = pcdep.get_link_args()
+                    self.version = pcdep.get_version()
+                    return
+            except Exception as e:
+                mlog.debug('Pcap not found via pkgconfig. Trying next, error was:', str(e))
+        if DependencyMethods.PCAPCONFIG in self.methods:
+            pcapconf = shutil.which('pcap-config')
+            if pcapconf:
+                stdo = Popen_safe(['pcap-config', '--cflags'])[1]
+                self.compile_args = stdo.strip().split()
+                stdo = Popen_safe(['pcap-config', '--libs'])[1]
+                self.link_args = stdo.strip().split()
+                self.version = '0'
```

Comment:
This is wrong, isn't it? It should at least match the 'fallback' version set by the EXTRAFRAMEWORK dependency later.

Alternatively, when there's no pkg-config dependency and we're not cross-compiling, we should do `self.compiler.run()` with code that runs `pcap_lib_version()` and fetches the version. But this is, of course, optional and we should try and talk to upstream about getting a pkg-config file (in addition to `pcap-config`).